app/controllers/favorite_tools_controller.rb

```diff
@@ -0,0 +1,40 @@
+class FavoriteToolsController < WithUserController
+  before_action :validate_ids
+
+  def index
+    render json: current_user_favorite_tools_response_json
+  end
+
+  def create
+    tool_ids.each do |tool_id|
+      current_user.favorite_tools.where(tool_id: tool_id).first_or_create
+    end
+    render json: current_user_favorite_tools_response_json
+  end
+
+  def destroy
+    current_user.favorite_tools.where(tool_id: tool_ids).delete_all
+    render json: current_user_favorite_tools_response_json
+  end
+
+  protected
+
+  def tool_ids
+    return [] unless params[:data] && params[:data].is_a?(Array)
+    return @tool_ids if @tool_ids
+
+    @tool_ids = params[:data].collect{ |data|
+      data[:id]
+    }
+  end
+
+  def validate_ids
+    missing = tool_ids - Resource.pluck(:id).collect(&:to_s)
+    error_response = missing.collect{ |tool_id| [tool_id, "invalid tool id"] }
+    render(json: { errors: error_response }) if error_response.any?
+  end
```

Comment:
something about this is brittle, if I send the `id` as a number it will fail, but if I send the same id as a string it will succeed.

I'd like this to not fail if the id is sent as a number.

Reply:
but if I use `"1"` it works correctly

Reply:
Oh, right, I'm used to the regular rails params that has all strings. These params are automatically parsed from the json so they can be ints.